src/main/scala/is/hail/annotations/StagedRegionValueBuilder.scala

```diff
@@ -0,0 +1,174 @@
+package is.hail.annotations
+
+import is.hail.asm4s.{AsmFunction2, Code, FunctionBuilder}
+import is.hail.asm4s.Code._
+import is.hail.asm4s._
+import is.hail.expr._
+import is.hail.utils._
+import org.objectweb.asm.tree.{AbstractInsnNode, IincInsnNode}
+
+import scala.collection.generic.Growable
+import scala.reflect.ClassTag
+
+class StagedRegionValueBuilder[T] private (val fb: FunctionBuilder[AsmFunction2[T, MemoryBuffer, Long]], val rowType: Type, var region: Code[MemoryBuffer], val pOffset: Code[Long])(implicit tti: TypeInfo[T]) {
```

Comment:
Let's remove the type parameter, `T`, it's only referenced by `val input` and that isn't used anywhere in this class. The type of `fb` can be `FunctionBuilder[_]`.

It would be nice to assert something about the function builder's second argument being a `MemoryBuffer`. It's a little tricky to do that with the current infrastructure. I'd just use `fb.getArg[MemoryBuffer](2)` and the programmer will get a run-time error if they screw up.

Reply:
Actually, I think we should make `MemoryBuffer` the first argument, that way multiple argument functions are just: `(MemoryBuffer, Arg1, Arg2, Arg3, ...) => Out`